tc/core/polyhedral/schedule_transforms.h

```diff
@@ -284,11 +284,21 @@ isl::multi_union_pw_aff partialScheduleMupa(
     const detail::ScheduleTree* root,
     const detail::ScheduleTree* tree);
 
-// Get the set of domain points active at the given node.  A domain
+// Get the set of domain points active at the given node, or below
+// the given node if "inclusive" is set.  A domain
 // point is active if it was not filtered away on the path from the
 // root to the node.  The root must be a domain element, otherwise no
 // elements would be considered active.
 isl::union_set activeDomainPoints(
+    const detail::ScheduleTree* root,
+    const detail::ScheduleTree* node,
+    bool inclusive = false);
```

Comment:
I am definitely not a fan of adding such boolean parameters to modify behavior.
Why not have activeDomainPoints inclusive (as it should already be because of my note above about ancestors). If you want exclusive behavior you can just do:
```
activeDomainPoints(root, node->ancestor(root, 1));
```
instead of the API naming bloat.

Reply:
I can rename `activeDomainPoints` to `activeDomainPointsAbove` and implement it in terms of `activeDomainPointsBelow`.  Changing `activeDomainPoints` to mean `activeDomainPointsBelow` does not sound like a good idea at all.  I'm already horribly confused by how the promotion performs the promotion underneath the node it is pointing at.